ui/app/ducks/gas/gas.duck.js

```diff
@@ -96,74 +85,70 @@ export function basicGasEstimatesLoadingFinished() {
 }
 
 async function basicGasPriceQuery() {
-  const url = `https://api.metaswap.codefi.network/gasPrices`
-  return await window.fetch(url, {
-    headers: {},
-    referrer: 'https://api.metaswap.codefi.network/gasPrices',
+  const url = 'https://api.metaswap.codefi.network/gasPrices'
+  const fetchOptions = {
+    referrer: url,
     referrerPolicy: 'no-referrer-when-downgrade',
-    body: null,
     method: 'GET',
     mode: 'cors',
-  })
+  }
+  const expirationOptions = { cacheRefreshTime: 75000 }
+
+  const responseJson = await fetchWithCache(
+    url,
+    fetchOptions,
+    expirationOptions,
+  )
+  return responseJson
 }
 
 export function fetchBasicGasEstimates() {
-  return async (dispatch, getState) => {
-    const { basicPriceEstimatesLastRetrieved } = getState().gas
-    const timeLastRetrieved =
-      basicPriceEstimatesLastRetrieved ||
-      (await getStorageItem('BASIC_PRICE_ESTIMATES_LAST_RETRIEVED')) ||
```

Comment:
Should we add a migration to remove these old cached values? :thinking: I'm not sure when they'd ever get cleared out otherwise.